domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -161,101 +208,340 @@ class TopicListController @Inject constructor(
+    if (nextChapterSummary != null) {
+      val numberOfDaysPassed = (
+        oppiaClock.getCurrentCalendar().timeInMillis -
+          lastCompletedChapterProgress.lastPlayedTimestamp
+        ) / ONE_DAY_IN_MS
+      val promotedStory = createPromotedStory(
+        storyId,
+        topic,
+        completedChapterProgressList.size,
+        story.chapterCount,
+        nextChapterSummary.name,
+        nextChapterSummary.explorationId
+      )
+      if (numberOfDaysPassed < ONE_WEEK_IN_DAYS) {
+        ongoingStoryListBuilder.addRecentStory(promotedStory)
+      } else {
+        ongoingStoryListBuilder.addOlderStory(promotedStory)
+      }
+    }
+  }
+
+  private fun createOngoingStoryListBasedOnRecentlyPlayed(
+    storyId: String,
+    story: StorySummary,
+    recentlyPlayerChapterProgress: ChapterProgress,
+    completedChapterProgressList: List<ChapterProgress>,
+    topic: Topic,
+    ongoingStoryListBuilder: OngoingStoryList.Builder
+  ) {
+    val recentlyPlayerChapterSummary: ChapterSummary? =
+      story.chapterList.find { chapterSummary ->
+        recentlyPlayerChapterProgress.explorationId == chapterSummary.explorationId
+      }
+    if (recentlyPlayerChapterSummary != null) {
+      val numberOfDaysPassed = (
+        oppiaClock.getCurrentCalendar().timeInMillis -
+          recentlyPlayerChapterProgress.lastPlayedTimestamp
+        ) / ONE_DAY_IN_MS
+      val promotedStory = createPromotedStory(
+        storyId,
+        topic,
+        completedChapterProgressList.size,
+        story.chapterCount,
+        recentlyPlayerChapterSummary.name,
+        recentlyPlayerChapterSummary.explorationId
+      )
+      if (numberOfDaysPassed < ONE_WEEK_IN_DAYS) {
+        ongoingStoryListBuilder.addRecentStory(promotedStory)
+      } else {
+        ongoingStoryListBuilder.addOlderStory(promotedStory)
+      }
+    }
+  }
+
+  private fun getStartedChapterProgressList(storyProgress: StoryProgress): List<ChapterProgress> {
+    return storyProgress.chapterProgressMap.values
+      .filter { chapterProgress ->
+        chapterProgress.chapterPlayState ==
+          ChapterPlayState.STARTED_NOT_COMPLETED
+      }
+      .sortedByDescending { chapterProgress -> chapterProgress.lastPlayedTimestamp }
+  }
+
+  private fun getCompletedChapterProgressList(storyProgress: StoryProgress): List<ChapterProgress> {
+    return storyProgress.chapterProgressMap.values
+      .filter { chapterProgress ->
+        chapterProgress.chapterPlayState ==
+          ChapterPlayState.COMPLETED
+      }
+      .sortedByDescending { chapterProgress -> chapterProgress.lastPlayedTimestamp }
+  }
+
+  private fun createRecommendedActivityList(
+    topicProgressList: List<TopicProgress>
+  ): RecommendedActivityList {
+    val recommendedActivityListBuilder = RecommendedActivityList.newBuilder()
+
+    if (topicProgressList.isNotEmpty()) {
+      val recommendedStoryBuilder = RecommendedStoryList.newBuilder()
+      if (topicProgressList.size == 1) {
```

Comment:
I'm not sure that I understand this special casing--why are we adding a suggested story when there's exactly one topic in-progress? I thought the plan was to show all recently played, then any other topics that are available to the learner up to 3 total tiles?

Reply:
]1) This case is to handle recommended stories only if any one topic is in progress and if there are more topic in progress it will show recently played stories list.
2) I misheard in our meeting that I need to show only one recommended story. I will fix this and show 3 recommended stories

Reply:
This seems like it can be simplified. Can't we populate a single RecommendedActivityList that includes up to 3 recommendations that are either:
- Recently played
- Suggestions
- Upcoming

Taken in that priority order? That was my understanding of what we should be showing. The header should change in the UI layer based on what data is provided (e.g. only recently played vs. a combination).

@rt4914 am I misremembering the design we discussed before the New Year?

Reply:
After chatting with @rt4914 a bit more, I came up with this almost-pseudocode: https://gist.github.com/BenHenning/0a67fd5dad7515065c07e2fec28a552b. Other than Rajat's comment, it *seems* like the desired approach can be simplified quite a bit more. @veena14cs could you please take another pass on the implementation here & consolidate it per the example above? If we can further simplify it then I think that will help more clearly show that it's implementing the desired behavior.

Reply:
(Please note that the sample above also isn't complete since it isn't handling the timestamp ordering bits, and is missing out other related functionality)